src/varintUtils.js

```diff
@@ -1,11 +1,22 @@
 import JSBI from "jsbi";
 
+const BIGINT_1 = JSBI.BigInt(1);
 const BIGINT_2 = JSBI.BigInt(2);
 
+export function interpretAsSignedType(n) {
+  // see https://github.com/protocolbuffers/protobuf/blob/master/src/google/protobuf/wire_format_lite.h#L857-L876
+  // TODO: does this handle sint64 as well?
```

Comment:
While the original implementation uses bit-wise operations for performance, I think here we can use a less confusing algorithm of

```
if (n is even) return n / 2;
else return -((n + 1) / 2);
```

Nevertheless I'm fine with this too if you verify that it works with 64-bit numbers.

Reply:
OK, I updated to use this algorithm instead.